File: ios/RTMPLiveStreamModule.swift
import ExpoModulesCore

public class RTMPLiveStreamModule: Module {
  public func definition() -> ModuleDefinition {
    
    Name("RTMPLiveStreamModule")

    AsyncFunction("startStreaming") { (options: [String: Any]) in
      RTMPLiveStreamManager.shared.start(options: options)
      return true
    }

    Function("stopStreaming") {
      RTMPLiveStreamManager.shared.stop()
    }

    Function("switchCamera") {
      RTMPLiveStreamManager.shared.switchCamera()
    }
  }
}

File: android/src/main/java/video/api/expo/rtmp/RTMPLiveStreamModule.kt
package video.api.expo.rtmp

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

class RTMPLiveStreamModule : Module() {

  override fun definition() = ModuleDefinition {
    Name("RTMPLiveStreamModule")

    // Start RTMP streaming
    AsyncFunction("startStreaming") { options: Map<String, Any> ->
      RTMPLiveStreamManager.start(options)
      true
    }

    Function("stopStreaming") {
      RTMPLiveStreamManager.stop()
    }

    Function("switchCamera") {
      RTMPLiveStreamManager.switchCamera()
    }
  }
}
